cms/djangoapps/contentstore/views/item.py

```diff
@@ -145,12 +146,19 @@ def xblock_handler(request, usage_key_string):
             accept_header = request.META.get('HTTP_ACCEPT', 'application/json')
 
             if 'application/json' in accept_header:
+                store = modulestore()
                 fields = request.GET.get('fields', '').split(',')
                 if 'graderType' in fields:
                     # right now can't combine output of this w/ output of _get_module_info, but worthy goal
                     return JsonResponse(CourseGradingModel.get_section_grader_type(usage_key))
+                if int(request.GET.get('course_tree', 0)):
+                    xblock = _get_xblock(usage_key, request.user)
+                    course = store.get_course(xblock.location.course_key)   # pylint: disable=no-member
+                    xblock_info = xblock_summary(xblock)
+                    course_outline = xblock_summary(course, include_children=True)
```

Comment:
Why does this API need to return both the course outline and the info about a given XBlock? It makes it into a confusing API, IMO. Presumably the move modal already has the XBlock info, so I think all it needs is the course outline. I would rather see a single course outline API that can be used by both the full course outline and by this new modal.